setup.py

```diff
@@ -1,120 +1,3 @@
-import os
-import re
-import shutil
-import sys
-from io import open
+from setuptools import setup
 
-from setuptools import find_packages, setup
-
-CURRENT_PYTHON = sys.version_info[:2]
-REQUIRED_PYTHON = (3, 8)
-
-# This check and everything above must remain compatible with Python 2.7.
-if CURRENT_PYTHON < REQUIRED_PYTHON:
-    sys.stderr.write("""
-==========================
-Unsupported Python version
-==========================
-
-This version of Django REST Framework requires Python {}.{}, but you're trying
-to install it on Python {}.{}.
-
-This may be because you are using a version of pip that doesn't
-understand the python_requires classifier. Make sure you
-have pip >= 9.0 and setuptools >= 24.2, then try again:
-
-    $ python -m pip install --upgrade pip setuptools
-    $ python -m pip install djangorestframework
-
-This will install the latest version of Django REST Framework which works on
-your version of Python. If you can't upgrade your pip (or Python), request
-an older version of Django REST Framework:
-
-    $ python -m pip install "djangorestframework<3.10"
-""".format(*(REQUIRED_PYTHON + CURRENT_PYTHON)))
-    sys.exit(1)
-
-
-def read(f):
-    with open(f, 'r', encoding='utf-8') as file:
-        return file.read()
-
-
-def get_version(package):
-    """
-    Return package version as listed in `__version__` in `init.py`.
-    """
-    init_py = open(os.path.join(package, '__init__.py')).read()
-    return re.search("__version__ = ['\"]([^'\"]+)['\"]", init_py).group(1)
-
-
-version = get_version('rest_framework')
-
-
-if sys.argv[-1] == 'publish':
-    if os.system("pip freeze | grep twine"):
-        print("twine not installed.\nUse `pip install twine`.\nExiting.")
-        sys.exit()
-    os.system("python setup.py sdist bdist_wheel")
-    if os.system("twine check dist/*"):
-        print("twine check failed. Packages might be outdated.")
-        print("Try using `pip install -U twine wheel`.\nExiting.")
-        sys.exit()
-    os.system("twine upload dist/*")
-    print("You probably want to also tag the version now:")
-    print("  git tag -a %s -m 'version %s'" % (version, version))
-    print("  git push --tags")
-    shutil.rmtree('dist')
-    shutil.rmtree('build')
-    shutil.rmtree('djangorestframework.egg-info')
-    sys.exit()
```

Comment:
Is there a maintainer doc/guide that needs updating? I found this page:

https://github.com/encode/django-rest-framework/blob/d3dd45b3f48856c3513ab1eb5354194fa9898f39/docs/community/project-management.md?plain=1#L75

Reply:
> Is it still up to date/relevant?

Invoking `setup.py` is deprecated and the available commands are being removed one by one, so I believe it needs updating

Reply:
OK, I'll update them

Reply:
Done